test/sagemaker_tests/pytorch/training/resources/mnist/smdataparallel_mnist.py

```diff
@@ -23,15 +24,23 @@
 from packaging.version import Version
 from torchvision import datasets, transforms
 from torch.optim.lr_scheduler import StepLR
-from torch.nn.parallel import DistributedDataParallel as DDP
-import torch.distributed as dist
-import smdistributed.dataparallel.torch.torch_smddp
 
-# set default instance type to p3.16
-if "SAGEMAKER_INSTANCE_TYPE" not in os.environ:
-    os.environ['SAGEMAKER_INSTANCE_TYPE'] = 'ml.p3.16xlarge'
+TORCH_VERSION = torch.__version__
 
-dist.init_process_group(backend='smddp')
+if Version(TORCH_VERSION) < Version("1.10"):
+    from smdistributed.dataparallel.torch.parallel.distributed import DistributedDataParallel as DDP
+    import smdistributed.dataparallel.torch.distributed as dist
+    dist.init_process_group()
+else:
+    from torch.nn.parallel import DistributedDataParallel as DDP
+    import torch.distributed as dist
+    import smdistributed.dataparallel.torch.torch_smddp
+
+    # set default instance type to p3.16
```

Comment:
why ml.p3 by default?

Reply:
+1

Reply:
AS stated here: https://github.com/aws/deep-learning-containers/pull/1948#discussion_r885216598 , I just merged the 2 files logically. I think the SMDataParallel team would have a better answer to it.